include/seqan3/search/algorithm/configuration/max_error.hpp

```diff
@@ -0,0 +1,204 @@
+    //!\}
+
+    /*!\brief A strong type of underlying type `uint8_t` or `double` that represents the number or rate of insertions.
+     * \tparam T The underlying type
+     * \ingroup search_configuration
+     */
+    template <typename T>
+    struct insertion : detail::strong_type<T, insertion<T>, detail::strong_type_skill::convert>
+    {
+        using detail::strong_type<T, insertion<T>, detail::strong_type_skill::convert>::strong_type;
+    };
+
+    /*!\name Template argument type deduction guides
+     * \relates seqan3::search_cfg::total
+     * \{
+     */
+    template <typename T>
+    insertion(T &&) -> insertion<T>;
+    //!\}
+
+    /*!\brief A strong type of underlying type `uint8_t` or `double` that represents the number or rate of deletions.
+     * \tparam T The underlying type
+     * \ingroup search_configuration
+     */
+    template <typename T>
+    struct deletion : detail::strong_type<T, deletion<T>, detail::strong_type_skill::convert>
+    {
+        using detail::strong_type<T, deletion<T>, detail::strong_type_skill::convert>::strong_type;
+    };
+
+    /*!\name Template argument type deduction guides
+     * \relates seqan3::search_cfg::deletion
+     * \{
+     */
+    template <typename T>
+    deletion(T &&) -> deletion<T>;
+    //!\}
+} // namespace seqan3::search_cfg
+
+namespace seqan3::detail
+{
+/*!\brief A configuration element for the maximum number of errors across all error types (mismatches, insertions,
+          deletions). This is an upper bound of errors independent from error numbers or rates of specific error types.
+ * \ingroup search_configuration
+ */
+struct search_config_max_error
+{
+    //!\brief The actual value.
+    std::tuple<uint8_t, uint8_t, uint8_t, uint8_t> value;
+};
+
+/*!\brief The max_error adaptor enabling pipe notation.
+ * \ingroup search_configuration
+ */
+struct search_config_max_error_adaptor : public configuration_fn_base<search_config_max_error_adaptor>
+{
+
+    /*!\brief Adds to the configuration a max_error configuration element.
+     * \relates seqan3::search_config_max_error
+     * \param[in] cfg  The configuration to be extended.
+     * \param[in] total_error The number of maximum errors over all error types.
+     * \param[in] substitution_error The number of maximum substitution errors.
+     * \param[in] insertion_error The number of maximum insertion errors.
+     * \param[in] deletion_error The number of maximum deletion errors.
+     * \returns A new configuration containing the max_error configuration element.
+     */
+    template <typename configuration_t>
+    //!\cond
+        requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
+    //!\endcond
+    // TODO: replace int by templates and check whether it can be casted to uint8_t
+    // TODO: template argument packing. allow any order and subset of strong types
+    constexpr auto invoke(configuration_t && cfg, search_cfg::total<int> const total_error,
```

Comment:
please use fixed size intege

Reply:
It will be replaced by uint8_t later but currently I run into problems with most fixed sized integers